backend/handlers/math/outcomes/dbpm/marketshares_test.go

```diff
@@ -0,0 +1,591 @@
+					MarketID: 1,
+				},
+			},
+			CoursePayouts: []CourseBetPayout{
+				{Payout: 0.25, Outcome: "YES"},
+				{Payout: -0.5, Outcome: "YES"},
+				{Payout: 0.25, Outcome: "NO"},
+				{Payout: -0, Outcome: "NO"}, // golang math.Round() rounds to -0 and +0
+				{Payout: 0.25, Outcome: "NO"},
+			},
+			F_YES:         0,
+			F_NO:          2,
+			ScaledPayouts: []int64{0, 0, 1, 0, 1},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualPayouts := CalculateScaledPayoutsDBPM(tc.Bets, tc.CoursePayouts, tc.F_YES, tc.F_NO)
+			if len(actualPayouts) != len(tc.ScaledPayouts) {
+				t.Fatalf("Test %s failed: expected %d payouts, got %d", tc.Name, len(tc.ScaledPayouts), len(actualPayouts))
+			}
+			for i, payout := range actualPayouts {
+				if payout != tc.ScaledPayouts[i] {
+					t.Errorf("Test %s failed at index %d: expected payout %d, got %d", tc.Name, i, tc.ScaledPayouts[i], payout)
+				}
+			}
+		})
+	}
+}
+
+func TestDivideUpMarketPoolSharesDBPM(t *testing.T) {
+	testcases := []struct {
+		Name               string
+		Bets               []models.Bet
+		ProbabilityChanges []wpam.ProbabilityChange
+		S_YES              int64
+		S_NO               int64
+	}{
+		{
+			Name: "PreventSimultaneousSharesHeld",
+			Bets: []models.Bet{
+				{
+					Amount:   3,
+					Outcome:  "YES",
+					Username: "user1",
+					PlacedAt: time.Date(2024, 5, 18, 5, 7, 31, 428975000, time.UTC),
+					MarketID: 3,
+				},
+				{
+					Amount:   1,
+					Outcome:  "NO",
+					Username: "user1",
+					PlacedAt: time.Date(2024, 5, 18, 5, 8, 13, 922665000, time.UTC),
+					MarketID: 3,
+				},
+			},
+			ProbabilityChanges: []wpam.ProbabilityChange{
+				{Probability: 0.5},
+				{Probability: 0.875},
+				{Probability: 0.7},
+			},
+			S_YES: 3,
+			S_NO:  1,
+		},
+		{
+			Name: "InfinityAvoidance",
+			Bets: []models.Bet{
+				{
+					Amount:   1,
+					Outcome:  "YES",
+					Username: "user2",
+					PlacedAt: now,
+					MarketID: 1,
+				},
+				{
+					Amount:   -1,
+					Outcome:  "YES",
+					Username: "user2",
+					PlacedAt: now.Add(time.Minute),
+					MarketID: 1,
+				},
+				{
+					Amount:   1,
+					Outcome:  "NO",
+					Username: "user1",
+					PlacedAt: now.Add(2 * time.Minute),
+					MarketID: 1,
+				},
+				{
+					Amount:   -1,
+					Outcome:  "NO",
+					Username: "user1",
+					PlacedAt: now.Add(3 * time.Minute),
+					MarketID: 1,
+				},
+				{
+					Amount:   1,
+					Outcome:  "NO",
+					Username: "user1",
+					PlacedAt: now.Add(4 * time.Minute),
+					MarketID: 1,
+				},
+			},
+			ProbabilityChanges: []wpam.ProbabilityChange{
+				{Probability: 0.50},
+				{Probability: 0.75},
+				{Probability: 0.50},
+				{Probability: 0.25},
+				{Probability: 0.50},
+				{Probability: 0.25},
+			},
+			S_YES: 0,
+			S_NO:  1,
+		},
+	}
+	ec := setup.EconomicsConfig()
+	ec.Economics.MarketCreation.InitialMarketSubsidization = 0
+	ec.Economics.MarketIncentives.CreateMarketCost = 1
+
```

Comment:
This is the test that was failing. These are the conditions that currently let it pass. After merging this PR, I have another branch in progress to allow dependency injection for these calculations; that will allow us to directly pass a configuration that is needed for this test.